### PR TITLE
[connectors]- fix(webcrawler): sanitize webcrawler url

### PR DESCRIPTION
**Pull request by JulesBelveze**
## Description

This PR aims to fix `invalid UTF8 null byte` errors in webcrawler URLs.

The issue occurs when URLs contain `//` that get encoded as null bytes when stored in Postgres. It adds URL validation and standardization to prevent these encoding errors

## Risk

Non negligible as it touches document upsert logic

### CODE REVIEW

File: connectors/src/connectors/webcrawler/temporal/activities.ts [hunk 1]
```diff
@@ -469,15 +470,22 @@ function formatDocumentContent({
 }): CoreAPIDataSourceDocumentSection {
   const URL_MAX_LENGTH = 128;
   const TITLE_MAX_LENGTH = 300;
-  const parsedUrl = new URL(url);
+
+  const validatedUrl = validateUrl(url);
+  if (!validatedUrl.valid || !validatedUrl.standardized) {
+    throw new Error(`Invalid URL: ${url}`);
```
Comment (spolu) [inline connectors/src/connectors/webcrawler/temporal/activities.ts:476]: Will this stop the workflow? We should just ignore here right?

Comment (spolu) [reply connectors/src/connectors/webcrawler/temporal/activities.ts:476]: Reading the code it seems it will not stop the workflow

Comment (JulesBelveze) [reply connectors/src/connectors/webcrawler/temporal/activities.ts:476]: Indeed, will skip in this case

File: connectors/src/connectors/webcrawler/temporal/activities.ts [hunk 2]
```diff
@@ -317,14 +318,26 @@ export async function crawlWebsiteByConnectorId(connectorId: ModelId) {
...
+              url: request.url,
+            });
+            if (!formattedDocumentContent) {
+              childLogger.info(
+                {
+                  documentId,
+                  configId: webCrawlerConfig.id,
+                  url,
+                },
+                `Invalid url. Skipping`
```
Comment (spolu) [inline connectors/src/connectors/webcrawler/temporal/activities.ts:333]: nit: I would align the log with the method called. `Invalid document or URL, skipping`

**Review by spolu — APPROVED** [review]
LGTM

**Review by flvndvd — APPROVED** [review]

**Pull request merged**